### PR TITLE
Support parsing and checking against tuple ATerms

### PR DESCRIPTION
**Pull request by Apanatshka**
This is purely a quick text-editor based change. I'd welcome any suggestions or additions to the PR to test that this (1) doesn't break anything and (2) actually works. 

Perhaps @pmisteliac can also review given that he made the most recent changes in `SPTUtil.java`?

### CODE REVIEW

**Review by Virtlink — COMMENTED** [review]
Did you test your addition? It looks fine, basically a modified copy of the constructor application match code. You could perhaps add a test for this new syntax, [maybe here](https://github.com/metaborg/spt/blob/master/org.metaborg.spt.test/parse/parse-to-aterm.spt) or in another of the SPT files.

File: org.metaborg.spt.core/src/main/java/org/metaborg/spt/core/SPTUtil.java [hunk 1]
```diff
@@ -226,6 +227,31 @@ public static boolean checkATermMatch(IStrategoTerm ast, IStrategoTerm match, IT
...
+                }
+                matchList = (IStrategoList) match.getSubterm(1);
+                if(ast.getSubtermCount() != matchList.size()) {
+                    logger.debug("The number of children {}, did not match the expected number {}",
+                        ast.getSubtermCount(), matchList.size());
+                    result = false;
+                    break;
+                }
+                matchIt = matchList.iterator();
+                stop = false;
```
Comment (pmisteliac) [inline org.metaborg.spt.core/src/main/java/org/metaborg/spt/core/SPTUtil.java:246]: I know it's used in the other cases as well, but I think stop is not a very good name.
Maybe `allSubtermsMatch` would be better

**Review by pmisteliac — APPROVED** [review]
Looks fine to me, maybe you could add a test case to the `org.metaborg.spt.test` project that covers the new case

**Comment by Apanatshka** [comment]
I'm the test I added now as part of the full Spoofax build. If everything checks out I will proceed to merge the branch.

**Review by Virtlink — APPROVED** [review]

**Pull request merged**